iree/compiler/Dialect/HAL/Target/LLVM/BUILD

```diff
@@ -11,15 +11,24 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
-load("//iree:build_defs.oss.bzl", "platform_trampoline_deps")
+
+load("//iree:build_defs.oss.bzl", "iree_cmake_extra_content", "platform_trampoline_deps")
 
 package(
     default_visibility = ["//visibility:public"],
     features = ["layering_check"],
     licenses = ["notice"],  # Apache 2.0
 )
 
+iree_cmake_extra_content(
+    content = """
+# if(NOT ${IREE_TARGET_BACKEND_LLVM-IR} AND NOT ${IREE_TARGET_BACKEND_DYLIB-LLVM-AOT})
```

Comment:
Delete comment?

Reply:
The `DYLIB-LLVM-AOT` target is not defined yet, but it should be used once added. This is sort of a TODO. Once @asaadaldien goes through and gets it enabled/working cleanly, we can update this.

The tricky detail here is that this file is used for both LLVM-IR and LLVM-AOT. To filter each separately we could

* Split directories into 'common', 'ir', and 'aot', continue to use `return()` as a way of filtering
* Instead of using `return()`, use `if() ... define targets ... endif()` for each (messier with `bazel_to_cmake`)

Reply:
I think splitting into the proposed directories is fine :)

Reply:
Updated TODO. I'd like to keep the directory move out of this change, but can prepare a PR for it.